services/federator/src/Federator/Service.hs

```diff
@@ -32,30 +30,31 @@ import Federator.Env (brig)
 import Imports
 import qualified Network.HTTP.Types as HTTP
 import Polysemy
+import Wire.API.Federation.GRPC.Types
 
-newtype BrigError = BrigErrorInvalidStatus HTTP.Status
+newtype ServiceError = ServiceErrorInvalidStatus HTTP.Status
   deriving (Eq, Show)
 
-data Brig m a where
+data Service m a where
   -- | Returns status and body, 'HTTP.Response' is not nice to work with in tests
-  BrigCall :: ByteString -> ByteString -> Domain -> Brig m (HTTP.Status, Maybe LByteString)
+  ServiceCall :: Component -> ByteString -> ByteString -> Domain -> Service m (HTTP.Status, Maybe LByteString)
 
-makeSem ''Brig
+makeSem ''Service
 
 -- FUTUREWORK(federation): Do we want to use servant client here? May make
 -- everything typed and safe
 --
 -- FUTUREWORK: Avoid letting the IO errors escape into `Embed Federator` and
 -- return them as `Left`
-interpretBrig ::
+interpretService ::
   Member (Embed Federator) r =>
-  Sem (Brig ': r) a ->
+  Sem (Service ': r) a ->
   Sem r a
-interpretBrig = interpret $ \case
-  BrigCall path body domain -> embed @Federator . liftAppIOToFederator $ do
+interpretService = interpret $ \case
+  ServiceCall component path body domain -> embed @Federator . liftAppIOToFederator $ do
     brigReq <- view brig <$> ask
     res <-
-      rpc' "brig" brigReq $
+      rpc' "brig" (brigReq component) $
         RPC.method HTTP.POST
           . RPC.path path -- FUTUREWORK(federation): Protect against arbitrary paths
           . RPC.body (RPC.RequestBodyBS body)
```

Comment:
I do not understand this part, but I see that renaming/generalizing in the PR from a brig to a service did not change this section much. This is not a Brig-specific request anymore, right? So perhaps `serviceReq` instead of `brigReq`? In the same vein, that `"brig"` string seems overly specific; I am not sure what exactly that argument to `rpc'` represents, but my understanding so far is that it is used only in printing/showing an exception, so how about using something like `show component` instead?